examples/Extensions/AddSitelinksUsingFeeds.cs

```diff
@@ -0,0 +1,366 @@
+            // remember what they are intended for later.
+            feed.Attributes.Add(CreateFeedAttribute("Link Text",
+                FeedAttributeType.String));
+            feed.Attributes.Add(CreateFeedAttribute("Link Final Url",
+                FeedAttributeType.UrlList));
+            feed.Attributes.Add(CreateFeedAttribute("Line 1",
+                FeedAttributeType.String));
+            feed.Attributes.Add(CreateFeedAttribute("Line 2",
+                FeedAttributeType.String));
+
+            FeedServiceClient feedServiceClient = client.GetService(Services.V3.FeedService);
+            FeedOperation operation = new FeedOperation()
+            {
+                Create = feed
+            };
+
+            MutateFeedsResponse response = feedServiceClient.MutateFeeds(
+                customerId.ToString(), new[] {operation});
+            string feedResourceName = response.Results[0].ResourceName;
+            Console.WriteLine($"Created feed with resource name '{feedResourceName}'.");
+
+            // After we create the feed, we need to fetch it so we can determine the
+            // attribute IDs, which will be required when populating feed items.
+            GoogleAdsServiceClient googleAdsService = client.GetService(
+                Services.V3.GoogleAdsService);
+            feed = googleAdsService.Search(
+                customerId.ToString(),
+                $"SELECT feed.attributes FROM feed WHERE feed.resource_name = '{feedResourceName}'"
+            ).First().Feed;
+
+            Hashtable attributeIds = new Hashtable
+            {
+                {"feed", feedResourceName},
+                // The attribute IDs come back in the same order that they were added.
+                {"linkTextAttributeId", feed.Attributes[0].Id},
+                {"finalUrlAttributeId", feed.Attributes[1].Id},
+                {"line1AttributeId", feed.Attributes[2].Id},
+                {"line2AttributeId", feed.Attributes[3].Id}
+            };
+
+            return attributeIds;
+        }
+
+        private FeedAttribute CreateFeedAttribute(string name, FeedAttributeType type)
+        {
+            return new FeedAttribute()
+            {
+                Name = name,
+                Type = type
+            };
+        }
+
+        private FeedItemOperation NewFeedItemOperation(Hashtable data, string text, string finalUrl,
+            string line1, string line2)
+        {
+            FeedItem feedItem = new FeedItem()
+            {
+                Feed = data["feed"].ToString()
+            };
+
+            FeedItemAttributeValue linkTextAttributeValue = new FeedItemAttributeValue()
+            {
+                FeedAttributeId = (long) data["linkTextAttributeId"],
+                StringValue = text
+            };
+            FeedItemAttributeValue finalUrlAttributeValue = new FeedItemAttributeValue()
+            {
+                FeedAttributeId = (long) data["finalUrlAttributeId"]
```

Comment:
I think you could use something like `StringValues = { finalUrl }` to make it inline.

Reply:
Agreed, fixed.